File: app/src/main/java/com/asf/wallet/di/TransactionDetailModule.java
package com.asf.wallet.di;

import com.asf.wallet.interact.FindDefaultNetworkInteract;
import com.asf.wallet.interact.FindDefaultWalletInteract;
import com.asf.wallet.repository.EthereumNetworkRepositoryType;
import com.asf.wallet.repository.WalletRepositoryType;
import com.asf.wallet.router.ExternalBrowserRouter;
import com.asf.wallet.viewmodel.TransactionDetailViewModelFactory;
import dagger.Module;
import dagger.Provides;

@Module public class TransactionDetailModule {

  @Provides TransactionDetailViewModelFactory provideTransactionDetailViewModelFactory(
      FindDefaultNetworkInteract findDefaultNetworkInteract,
      FindDefaultWalletInteract findDefaultWalletInteract,
      ExternalBrowserRouter externalBrowserRouter) {
    return new TransactionDetailViewModelFactory(findDefaultNetworkInteract,
        findDefaultWalletInteract, externalBrowserRouter);
  }

  @Provides FindDefaultNetworkInteract provideFindDefaultNetworkInteract(
      EthereumNetworkRepositoryType ethereumNetworkRepository) {
    return new FindDefaultNetworkInteract(ethereumNetworkRepository);
  }

  @Provides ExternalBrowserRouter externalBrowserRouter() {
    return new ExternalBrowserRouter();
  }

  @Provides FindDefaultWalletInteract findDefaultWalletInteract(
      WalletRepositoryType walletRepository) {
    return new FindDefaultWalletInteract(walletRepository);
  }
}

File: app/src/main/java/com/asfoundation/wallet/ui/balance/detail/TransactionDetailModule.kt
package com.asfoundation.wallet.ui.balance.detail

import com.appcoins.wallet.bdsbilling.WalletService
import com.asfoundation.wallet.C
import com.asfoundation.wallet.interact.FindDefaultNetworkInteract
import com.asfoundation.wallet.router.ExternalBrowserRouter
import com.asfoundation.wallet.router.TransactionDetailRouter
import com.asfoundation.wallet.support.SupportInteractor
import dagger.Module
import dagger.Provides
import io.reactivex.disposables.CompositeDisposable

@Module
class TransactionDetailModule {
  @Provides
  fun provideTransactionDetailViewModelFactory(
      data: TransactionDetailData,
      interactor: TransactionDetailInteractor,
      walletService: WalletService,
      findDefaultNetworkInteract: FindDefaultNetworkInteract,
      externalBrowserRouter: ExternalBrowserRouter, supportInteractor: SupportInteractor,
      transactionDetailRouter: TransactionDetailRouter): TransactionDetailViewModelFactory {
    return TransactionDetailViewModelFactory(data, interactor, findDefaultNetworkInteract,
        walletService, externalBrowserRouter, CompositeDisposable(), supportInteractor,
        transactionDetailRouter)
  }

  @Provides
  fun provideTransactionDetailRouter() = TransactionDetailRouter()

  @Provides
  fun providesTransactionDetailData(activity: TransactionDetailActivity): TransactionDetailData {
    return TransactionDetailData(activity.intent.getParcelableExtra(C.Key.TRANSACTION))
  }

  @Provides
  fun externalBrowserRouter() = ExternalBrowserRouter()
}